src/IssueViz.Security.UnitTests/OpenInIDE/Api/OpenInIDEFailureInfoBarTests.cs

```diff
@@ -0,0 +1,226 @@
+                .SetupSequence(x => x.AttachInfoBarWithButton(HotspotsToolWindow.ToolWindowId, It.IsAny<string>(), It.IsAny<string>(), default))
+                .Returns(firstInfoBar.Object)
+                .Returns(secondInfoBar.Object);
+
+            var testSubject = new OpenInIDEFailureInfoBar(infoBarManager.Object, Mock.Of<IOutputWindowService>());
+
+            // Act
+            testSubject.Show(); // show first bar
+            testSubject.Show(); // show second bar
+
+            firstInfoBar.VerifyNoOtherCalls();
+            secondInfoBar.VerifyNoOtherCalls();
+            infoBarManager.VerifyAll();
+            infoBarManager.Verify(x => x.DetachInfoBar(firstInfoBar.Object), Times.Once);
+            infoBarManager.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void Dispose_HasPreviousInfoBar_InfoBarRemoved()
+        {
+            var infoBar = new Mock<IInfoBar>();
+            var infoBarManager = new Mock<IInfoBarManager>();
+            var testSubject = CreateTestSubjectWithPreviousInfoBar(infoBarManager, infoBar);
+
+            SetupInfoBarEvents(infoBar);
+
+            // Act
+            testSubject.Dispose();
+
+            CheckInfoBarWithEventsRemoved(infoBarManager, infoBar);
+            infoBarManager.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void Dispose_NoPreviousInfoBar_NoException()
+        {
+            var infoBarManager = new Mock<IInfoBarManager>();
+            var testSubject = new OpenInIDEFailureInfoBar(infoBarManager.Object, Mock.Of<IOutputWindowService>());
+
+            testSubject.Dispose();
+
+            infoBarManager.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void InfoBarIsManuallyClosed_InfoBarDetachedFromToolWindow()
+        {
+            var infoBar = new Mock<IInfoBar>();
+            var infoBarManager = new Mock<IInfoBarManager>();
+            var testSubject = CreateTestSubjectWithPreviousInfoBar(infoBarManager, infoBar);
+
+            SetupInfoBarEvents(infoBar);
+
+            // Act
+            infoBar.Raise(x => x.Closed += null, EventArgs.Empty);
+
+            CheckInfoBarWithEventsRemoved(infoBarManager, infoBar);
+            infoBarManager.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void InfoBarButtonClicked_OutputWindowIsShown()
+        {
+            var infoBar = new Mock<IInfoBar>();
+            var outputWindowService = new Mock<IOutputWindowService>();
+            var testSubject = CreateTestSubjectWithPreviousInfoBar(infoBar: infoBar, outputWindow: outputWindowService);
+
+            outputWindowService.VerifyNoOtherCalls();
+
+            // Act
+            infoBar.Raise(x => x.ButtonClick += null, EventArgs.Empty);
+
+            outputWindowService.Verify(x => x.Show(), Times.Once);
+            outputWindowService.VerifyNoOtherCalls();
+        }
+
+        private static OpenInIDEFailureInfoBar CreateTestSubjectWithPreviousInfoBar(
```

Comment:
The code in this test appeared multiple times in the old tests so I extracted it to a helper method.